lighthouse-extension/test/extension.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+const path = require('path');
+const assert = require('assert');
+const puppeteer = require('puppeteer');
+
+const lighthouseExtensionPath = path.resolve(__dirname, '../app');
+const config = require(path.resolve(__dirname, '../../lighthouse-core/config/default.js'));
+
+puppeteer.launch({
+  headless: false,
+  executablePath: process.env.CHROME_PATH ? process.env.CHROME_PATH : undefined,
+  args: [
+    `--disable-extensions-except=${lighthouseExtensionPath}`,
+    `--load-extension=${lighthouseExtensionPath}`,
+  ],
+}).then(browser => {
+  browser.newPage()
+    .then(page => {
+      return page.goto('https://www.paulirish.com', {waitUntil: 'networkidle2'});
+    })
+    .then(() => browser.targets())
+    .then(targets => {
+      const extensionTarget = targets.find(target => {
+        return target._targetInfo.title === 'Lighthouse';
+      });
+
+      if (extensionTarget) {
+        extensionTarget.createCDPSession()
```

Comment:
Any way to use puppeteer's `evaluate` method(s) here 

https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#executioncontextevaluatepagefunction-args maybe?

```js
await ctx.evaluate(() => {
  return runLighthouseInExtension({restoreCleanState: true}, ['performance', 'pwa', 'accessibility', 'best-practices', 'seo']);
});
```

Whatever that looks like. Not sure if you can get the crx context, other than what you're doing using an private api.

Reply:
sounds neat, will try :)

Reply:
yup +1 to using this rather than dropping to the protocol. good call.

Reply:
@ebidel not sure how I can get that execution context from the Target class. I tried `.page().mainFrame().executionContext()` but page() is of course null.

Any ideas? 😛

Reply:
Asking...